rewrite-java/src/main/java/org/openrewrite/java/cleanup/HideUtilityClassConstructorVisitor.java

```diff
@@ -0,0 +1,303 @@
+                                c.getBody().getCoordinates().lastStatement(),
+                                classDecl.getSimpleName()
+                        ), p);
+            }
+            return c;
+        }
+    }
+
+    /**
+     * We consider a Utility Class to have an "exposed" constructor if the constructor is Public or Package-Private.
+     * The constructor may be "Protected" in cases where it's desirable to subclass the Utility Class.
+     */
+    private static class UtilityClassWithExposedConstructorInspectionVisitor<P> extends JavaIsoVisitor<P> {
+        public UtilityClassWithExposedConstructorInspectionVisitor() {
+            setCursoringOn();
+        }
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, P p) {
+            J.MethodDeclaration md = super.visitMethodDeclaration(method, p);
+            if (md.isConstructor() && !(md.hasModifier(J.Modifier.Type.Private) || md.hasModifier(J.Modifier.Type.Protected))) {
+                // If visibility is Public, replace it with Private
+                md = maybeAutoFormat(md,
+                        md.withModifiers(
+                                ListUtils.map(md.getModifiers(), mod -> mod.getType() == J.Modifier.Type.Public ? mod.withType(J.Modifier.Type.Private) : mod)
+                        ), p, getCursor().dropParentUntil(J.class::isInstance));
+                // If visibility is Package-Private (no access modifier keyword), replace it with Private
+                if (!md.hasModifier(J.Modifier.Type.Private)) {
+                    J.Modifier mod = new J.Modifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, J.Modifier.Type.Private);
+                    md = maybeAutoFormat(md,
+                            md.withModifiers(
+                                    ListUtils.concat(md.getModifiers(), mod)
+                            ), p, getCursor().dropParentUntil(J.class::isInstance));
+                }
+            }
+            return md;
+        }
+    }
+
+    /**
+     * Utility class for identifying utility classes.
+     * <p>
+     * A Class is considered a Utility Class if it meets these rules:
+     *
+     * <ul>
+     *     <li>A utility class contains only static methods and fields in its API</li>
+     *     <li>A utility class does not have any public constructors</li>
+     *     <li>A utility class does not have any {@code implements} or {@code extends} keywords.</li>
+     * </ul>
+     *
+     * <p>
+     * What's an actual example of a utility class, you ask? Well, here's one.
+     * <p>
+     * We are keeping this class private for the moment in order to keep the API surface area small. However,
+     * there are other cleanup rules which could benefit from these, such as "make utility class final", etc.
+     * Until then, however, we'll keep this private and unexposed.
+     */
+    private static final class UtilityClassUtilities {
```

Comment:
Can move this into its own class & making it package-private. Or have these be members of the visitor. Trying to keep this encapsulated but also as hidden as possible for the time being